source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -57,6 +82,17 @@ enum uma_result_t umaMemoryProviderAlloc(uma_memory_provider_handle_t hProvider,
 enum uma_result_t umaMemoryProviderFree(uma_memory_provider_handle_t hProvider,
                                         void *ptr, size_t size);
 
+///
+/// \brief Set attributes of the memory space pointed by ptr from the memory provider
+/// \param hProvider handle to the memory provider
+/// \param ptr pointer to the allocated memory
+/// \param size size of the allocation
+/// \param attrs attributes of the memory space to be set
+///
+enum uma_result_t
+umaMemoryProviderSetAttributes(uma_memory_provider_handle_t hProvider,
```

Comment:
I have a feeling that we are mixing madvise-like attributes (just hints that runtime can ignore) with flags that should be applied or an error should be returned. Is it the right approach? I think we should have a discussion when we introduce such API.

Reply:
But can we define a generic set of flags vs attributes for all providers?

Reply:
BTW, we need only `MADV_DONTNEED, MADV_HUGEPAGE, MADV_NOHUGEPAGE` in qkmalloc.